api/bazel/api_build_system.bzl

```diff
@@ -24,21 +24,39 @@ def _LibrarySuffix(library_name, suffix):
 # https://github.com/bazelbuild/bazel/issues/3935 and/or
 # https://github.com/bazelbuild/bazel/issues/2626 are resolved.
 def api_py_proto_library(name, srcs = [], deps = [], has_services = 0):
-    py_proto_library(
+    _py_proto_library(
         name = _Suffix(name, _PY_SUFFIX),
         srcs = srcs,
         default_runtime = "@com_google_protobuf//:protobuf_python",
         protoc = "@com_google_protobuf//:protoc",
         deps = [_LibrarySuffix(d, _PY_SUFFIX) for d in deps] + [
             "@com_envoyproxy_protoc_gen_validate//validate:validate_py",
-            "@googleapis//:api_httpbody_protos_py",
-            "@googleapis//:http_api_protos_py",
-            "@googleapis//:rpc_status_protos_py",
+            "@com_google_googleapis//google/rpc:status_py_proto",
+            "@com_google_googleapis//google/api:annotations_py_proto",
+            "@com_google_googleapis//google/api:http_py_proto",
+            "@com_google_googleapis//google/api:httpbody_py_proto",
             "@com_github_gogo_protobuf//:gogo_proto_py",
         ],
         visibility = ["//visibility:public"],
     )
 
+# This replaces googleapis rules.
```

Comment:
Unclear to me what this means. Can you expand?

Reply:
Reading later parts of the PR, I get what it means, but this isn't a standalone comment. Also, why do we even need to do this?

Reply:
There is no public py_proto_library (https://github.com/grpc/grpc/issues/19255), so the rules in googleapis for python are not defined. This provides an implementation using protobuf repository rules.

Reply:
It's a bit confusing that we say we are replacing something that is not defined :) I think the ask from my end here is to have a clear comment explaining the full story. More detail is better. Thanks.

Reply:
Done